lisa/nic.py

```diff
@@ -5,116 +5,76 @@
 import os
 import re
 from collections import OrderedDict
+from dataclasses import dataclass
 from pathlib import PurePosixPath
 from typing import TYPE_CHECKING, Any, Dict, List, Optional
 
 from assertpy import assert_that
 from retry import retry
 
-from lisa.tools import Ip, Lspci, Tee
-from lisa.util import InitializableMixin, LisaException, constants, find_groups_in_lines
+from lisa.tools import Cat, Ip, KernelConfig, Lspci, Modprobe, Tee
+from lisa.util import InitializableMixin, LisaException, constants
 
 if TYPE_CHECKING:
     from lisa import Node
 
 
 class NicInfo:
-    # Class for info about an single upper/lower nic pair.
-    # Devices using SRIOV on azure typically have an upper synthetic device
-    # paired with a lower SRIOV Virtual Function (VF) device that
+    # Class for info about an single nic/pci device pair.
+    # Devices using AN on azure typically have an network interface (NIC).
+    # paired with a PCI device that
     # enables the passthrough to the physical NIC.
 
-    # If sriov(AN) is not enabled then there will not be a lower SRIOV VF.
+    # If AN is not enabled then there will not be a pci device.
     # In this case, NicInfo will have lower = ""
 
     def __init__(
         self,
-        upper: str,
+        name: str,
         lower: str = "",
         pci_slot: str = "",
+        lower_module_name: str = "",
+        driver_sysfs_path: Optional[PurePosixPath] = None,
     ) -> None:
-        self.upper = upper
+        self.name = name
         self.lower = lower
-        self.pci_slot = pci_slot
-        self.ip_addr = ""
         self.mac_addr = ""
+        self.ip_addr = ""
+        self.pci_slot = pci_slot
         self.dev_uuid = ""
-        self.bound_driver = ""
-        self.driver_sysfs_path = PurePosixPath("")
-        self.numa_node = 0
+        self.module_name = ""
+        if driver_sysfs_path is None:
+            self.driver_sysfs_path = PurePosixPath("")
+        else:
+            self.driver_sysfs_path = driver_sysfs_path
+        self.lower_module_name = lower_module_name
 
     def __str__(self) -> str:
         return (
             "NicInfo:\n"
-            f"upper: {self.upper}\n"
-            f"lower: {self.lower}\n"
+            f"name: {self.name}\n"
             f"pci_slot: {self.pci_slot}\n"
             f"ip_addr: {self.ip_addr}\n"
             f"mac_addr: {self.mac_addr}\n"
         )
 
+    @property
+    def is_pci(self) -> bool:
+        return len(self.lower) == 0 and len(self.pci_slot) > 0
```

Comment:
What does mean pci here? My understanding is the synthetic nic with VF is also a PCI. If not, how to present the lower pci device?

Reply:
I want to use it, there if no VF, but synthetic nic has AN capabilities here.